### PR TITLE
Add gmsaProfile in WindowsProfile for AKS

### PR DESCRIPTION
**Pull request by AbelHu**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month. 
  September 2021
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month. 
  September 2021
  4. If updating an existing version, please select the specific langauge SDKs and CLIs that must be refreshed after the swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following apply to the PR so that label “WaitForARMFeedback” will be added automatically to begin ARM API Review. Failure to comply may result in delays to the manifest.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[x] To review changes efficiently, ensure you copy the existing version into the new directory structure for first commit and then push new changes, including version updates, in separate commits.

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2021-09-01/managedClusters.json [hunk 1]
```diff
@@ -2929,13 +2932,35 @@
...
         "adminUsername"
       ],
       "description": "Profile for Windows VMs in the managed cluster."
     },
+    "WindowsGmsaProfile": {
+      "properties": {
+        "enabled": {
+          "type": "boolean",
+          "title": "Whether to enable Windows gMSA.",
+          "description": "Specifiy whether to enable Windows gMSA in the managed cluster."
```
Comment (chengliangli0918) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2021-09-01/managedClusters.json:2951]: typo

File: specification/containerservice/resource-manager/readme.java.md [hunk 2]
```diff
@@ -38,6 +38,20 @@ batch:
   - tag: package-2021-05
```
Comment (chengliangli0918) [inline specification/containerservice/resource-manager/readme.java.md:38]: also need to update readme.md, refer to the previous APIs

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2021-09-01/managedClusters.json [hunk 3]
```diff
@@ -2929,13 +2932,35 @@
           "type": "boolean",
           "title": "Whether to enable CSI proxy.",
           "description": "For more details on CSI proxy, see the [CSI proxy GitHub repo](https://github.com/kubernetes-csi/csi-proxy)."
+        },
+        "gmsaProfile": {
+          "$ref": "#/definitions/WindowsGmsaProfile",
+          "description": "The Windows gMSA Profile in the Managed Cluster."
```
Comment (chengliangli0918) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2021-09-01/managedClusters.json:2938]: need to add Gmsa to custom-words.txt to fix spell check

File: specification/containerservice/resource-manager/readme.java.md (hunk 2 above)
Comment (AbelHu) [reply specification/containerservice/resource-manager/readme.java.md:38]: Weird. I did update readme.md with below content but it does not exist in this PR.
```

### Tag: package-2021-09

These settings apply only when `--tag=package-2021-09` is specified on the command line.

```yaml $(tag) == 'package-2021-09'
input-file:
  - Microsoft.ContainerService/stable/2021-09-01/managedClusters.json
```
You can see https://github.com/AbelHu/azure-rest-api-specs/commit/34d6e83e8145836bc31e342b7eaed3aa95580765#diff-5edcad5257543ceda29d72e1ff8cfb550ced7ae4e844feca5f1f591214a7ce89

Comment (AbelHu) [reply specification/containerservice/resource-manager/readme.java.md:38]: Found the reason. The file was automatically updated when creating the branch dev-containerservice-Microsoft.ContainerService-2021-09-01. https://github.com/Azure/azure-rest-api-specs/blob/dev-containerservice-Microsoft.ContainerService-2021-09-01/specification/containerservice/resource-manager/readme.md

**Comment by AbelHu** [comment]
@chengliangli0918 do you have any thoughts for the failures ❌ azure-sdk-for-python-track2 failed [Detail] and ️❌ azure-sdk-for-go-track2 failed [Detail]?

**Comment by chengliangli0918** [comment]
@AbelHu  in dev-containerservice-Microsoft.ContainerService-2021-08-01 branch, for all commits PRs and final merging PR to main, azure-sdk-for-python-track2 and azure-sdk-for-go-track2 didn't fail. Maybe you can check with @ArcturusZhang for some insights.

> @chengliangli0918 do you have any thoughts for the failures ❌ azure-sdk-for-python-track2 failed [Detail] and ️❌ azure-sdk-for-go-track2 failed [Detail]?

**Comment by AbelHu** [comment]
Thanks @ArcturusZhang for confirming that this is caused by that this PR is not targeting main branch so we can ignore it. cc @chengliangli0918

**Comment by AbelHu** [comment]
@chengliangli0918 could you help to approve this PR so @ArcturusZhang can help to merge this PR to the dev branch?

**Review by chengliangli0918 — APPROVED** [review]

**Comment by chengliangli0918** [comment]
@AbelHu I approved this PR. However you checked options in description which triggers 'WaitForARMFeedback' tag, so you need signoff from ARM team, JFYI. 

> @chengliangli0918 could you help to approve this PR so @ArcturusZhang can help to merge this PR to the dev branch?

File: specification/containerservice/resource-manager/readme.java.md (hunk 2 above)
Comment (chengliangli0918) [reply specification/containerservice/resource-manager/readme.java.md:38]: Though the readme.md was automatically updated when creating the branch 2021-09-01 branch, we still need update it with a few changes, see my commit https://github.com/Azure/azure-rest-api-specs/pull/15476 for reference.

**Review by j5lim — APPROVED** [review]
Approved from ARM side.

**Review by chengliangli0918 — COMMENTED** [review]
reame.md needs some update

**Comment by AbelHu** [comment]
> reame.md needs some update

Thanks for the catching, @chengliangli0918! Updated readme.md

**Review by ArcturusZhang — APPROVED** [review]

**Pull request merged**